src/main/java/org/apache/datasketches/kll/KllDirectDoublesSketch.java

```diff
@@ -0,0 +1,326 @@
+
+import org.apache.datasketches.Family;
+import org.apache.datasketches.memory.MemoryRequestServer;
+import org.apache.datasketches.memory.WritableMemory;
+
+/**
+ * This class implements an off-heap doubles KllSketch via a WritableMemory instance of the sketch.
+ *
+ * <p>Please refer to the documentation in the package-info:<br>
+ * {@link org.apache.datasketches.kll}</p>
+ *
+ * @author Lee Rhodes, Kevin Lang
+ */
+public final class KllDirectDoublesSketch extends KllDirectSketch {
+
+  /**
+   * The actual constructor.
+   * @param wmem the current WritableMemory
+   * @param memReqSvr the given MemoryRequestServer to request a larger WritableMemory
+   * @param memVal the MemoryValadate object
+   */
+  private KllDirectDoublesSketch(final WritableMemory wmem, final MemoryRequestServer memReqSvr,
+      final MemoryValidate memVal) {
+    super(SketchType.DOUBLES_SKETCH, wmem, memReqSvr, memVal);
+  }
+
+  /**
+   * Wrap a sketch around the given source Memory containing sketch data that originated from
+   * this sketch.
+   * @param srcMem a WritableMemory that contains data.
+   * @param memReqSvr the given MemoryRequestServer to request a larger WritableMemory
+   * @return instance of this sketch
+   */
+  public static KllDirectDoublesSketch writableWrap(final WritableMemory srcMem, final MemoryRequestServer memReqSvr) {
+    final MemoryValidate memVal = new MemoryValidate(srcMem);
+    return new KllDirectDoublesSketch(srcMem, memReqSvr, memVal);
+  }
+
+  /**
+   * Create a new instance of this sketch using default M.
+   * @param k parameter that controls size of the sketch and accuracy of estimates
+   * @param dstMem the given destination WritableMemory object for use by the sketch
+   * @param memReqSvr the given MemoryRequestServer to request a larger WritableMemory
+   * @return a new instance of this sketch
+   */
+  public static KllDirectDoublesSketch newInstance(final int k, final WritableMemory dstMem,
+      final MemoryRequestServer memReqSvr) {
+    return newInstance(k, DEFAULT_M, dstMem, memReqSvr);
+  }
+
+  /**
+   * Create a new instance of this sketch.
+   * @param k parameter that controls size of the sketch and accuracy of estimates
+   * @param m parameter that controls the minimum level width.
+   * @param dstMem the given destination WritableMemory object for use by the sketch
+   * @param memReqSvr the given MemoryRequestServer to request a larger WritableMemory
+   * @return a new instance of this sketch
+   */
+  public static KllDirectDoublesSketch newInstance(final int k, final int m, final WritableMemory dstMem,
+      final MemoryRequestServer memReqSvr) {
+    insertPreInts(dstMem, PREAMBLE_INTS_DOUBLE);
+    insertSerVer(dstMem, SERIAL_VERSION_UPDATABLE);
+    insertFamilyID(dstMem, Family.KLL.getID());
+    insertFlags(dstMem, DOUBLES_SKETCH_BIT_MASK | UPDATABLE_BIT_MASK);
+    insertK(dstMem, k);
+    insertM(dstMem, m);
+    insertN(dstMem, 0);
+    insertDyMinK(dstMem, k);
```

Comment:
I cannot guess from the name what this is

Reply:
Before, we had two different variables named minK ... and they just happened to not collide.  Now however, due to the changes I recently made in enabling _m_  the static minK has gone away.  I could change this one back to minK, or expand its name to "dynamicMinK".  Preference?

Reply:
Fixed from earlier comment.

Done.

Reply:
I would vote for just minK